src/MICore/Transports/LocalLinuxTransport.cs

```diff
@@ -187,9 +195,29 @@ public override void InitStreams(LaunchOptions options, out StreamReader reader,
             reader = new StreamReader(gdbStdOutStream);
         }
 
+        private void FifoWatcher_Deleted(object sender, FileSystemEventArgs e)
+        {
+            if (e.FullPath == _gdbStdInName || e.FullPath == _gdbStdOutName)
+            {
+                Logger?.WriteLine("Gdb fifo deleted, stop debugging");
+                _fifoWatcher.Deleted -= FifoWatcher_Deleted;
+                this.Callback.OnDebuggerProcessExit(null);
+            }
+        }
+
         protected override string GetThreadName()
         {
             return "MI.LocalLinuxTransport";
         }
+
+        public override void Close()
+        {
+            base.Close();
```

Comment:
I double checked this, there won't be a race condition. When user stop debugging from the UI, we either send "kill" command or "exec-detach", which will make the thread-group exit which will in turn cause the transport to Close(). and the file watcher will be removed.
After all this, we'll send "exec-quit" to quit Gdb and only then the fifos will be removed.